protons/driver.py

```diff
@@ -587,6 +588,10 @@ def _set_titration_state(self, titration_group_index, titration_state_index, con
                             (titration_state_index, self._get_num_titration_states(titration_group_index)))
 
         self._update_forces(titration_group_index, titration_state_index, context=context)
+        # Update paramters in context
```

Comment:
For efficiency, I moved `updateParametersInContext()` to after all parameters have been modified. Now, it is only called once per attempt.